src/entities/trades/gnosis-protocol/CoWTrade.ts

```diff
@@ -206,16 +185,12 @@ export class CoWTrade extends Trade {
         sellToken: tokenIn.address,
       })
 
-      // Calculate the fee in terms of percentages
-      const feeAmountBN = parseUnits(quoteResponse.quote.feeAmount.toString(), tokenIn.decimals)
-        .div(quoteResponse.quote.sellAmount.toString())
-        .mul(100)
-      const tokenInDenominator = parseUnits('100', tokenIn.decimals).toBigInt()
-      const fee = new Percent(feeAmountBN.toBigInt(), tokenInDenominator)
+      // CoW Swap doesn't charge any fee
+      const fee = new Percent(ZERO, ONE)
 
       const feeAmount = Currency.isNative(currencyAmountIn.currency)
-        ? CurrencyAmount.nativeCurrency(quoteResponse.quote.feeAmount.toString(), chainId)
-        : new TokenAmount(currencyAmountIn.currency as Token, quoteResponse.quote.feeAmount.toString())
+        ? CurrencyAmount.nativeCurrency('0', chainId)
```

Comment:
MINOR, non-blocking
Maybe we can use the newly imported `ZERO` bigInt instead of all these `'0'` strings, as these functions take the `BigIntish` type. I think it should work and we would drop the hardcoded string values. I didn't try this out tho.

Reply:
Good catch, done!